python/ray/autoscaler/_private/commands.py

```diff
@@ -294,6 +310,30 @@ def handle_cli_override(key, override):
         override_cluster_name,
         no_monitor_on_head,
     )
+
+    if wait_for_workers:
+        WAIT_FOR_WORKERS_DELAY = 10
+        provider = _get_node_provider(config["provider"], config["cluster_name"])
+
+        # Wait for worker nodes to launch
+        time.sleep(WAIT_FOR_WORKERS_DELAY)
```

Comment:
This strategy seems fine as a best-effort attempt to wait for workers to come up -- at the very least, the function call would exit at some point at or before the time the workers are ready.

But there are also some conditions under this will exit before the workers are ready.
For example, it could conceivably happen that pending workers are not registered within 10 seconds of the head's creation.
Or it could happen that there are some failed launches, then this method exits, and then the autoscaler tries to re-launch.

Another strategy is to wait for a user-specified number of workers to register in "STATUS_UP_TO_DATE" state -- that might be closer to what you're looking for. A slight refinement of that is to read the desired min_workers from the autoscaling config and wait for that many workers. (I'm assuming you're trying to block until resources are actually available.)

Reply:
In general, it's not great to have a sleep(n) and rely on something happening within n seconds.

Reply:
Thanks! From what I see, it looks like min_workers is no longer exposed as a global configuration variable but rather defined on a per node-type basis. Therefore, going with a specified number of worker nodes as an argument to the flag makes more sense to me.